tests/src/system/basic/WskActionTests.scala

```diff
@@ -0,0 +1,268 @@
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "pass parameters bound on creation-time to the action" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "printParams"
+            val params = Map(
+                "param1" -> "test1",
+                "param2" -> "test2")
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) =>
+                    action.create(
+                        name,
+                        Some(TestUtils.getCatalogFilename("samples/printParams.js")),
+                        parameters = params.mapValues(_.toJson))
+            }
+
+            val invokeParams = Map("payload" -> testString)
+            val run = wsk.action.invoke(name, invokeParams.mapValues(_.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    val logs = activation.logs.get.mkString(" ")
+
+                    (params ++ invokeParams).foreach {
+                        case (key, value) =>
+                            logs should include(s"params.$key: $value")
+                    }
+            }
+    }
+
+    it should "copy an action and invoke it successfully" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "copied"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some("/whisk.system/samples/wordCount"), Some("copy"))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "recreate and invoke a new action with different code" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "recreatedAction"
+            assetHelper.withCleaner(wsk.action, name, false) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/wc.js")))
+            }
+
+            val run1 = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run1) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.logs.get.mkString(" ") should include(s"The message '$testString' has")
+            }
+
+            wsk.action.delete(name)
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/hello.js")))
```

Comment:
Since I am refactoring all the test cases to remove the catalog, could you change getCatalogFilename into getTestActionFilename? If the script is not available under tests/dat/actions, please add it.